centreon/src/Core/Connector/Application/UseCase/FindConnectors/FindConnectors.php

```diff
@@ -0,0 +1,175 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\Connector\Application\UseCase\FindConnectors;
+
+use Centreon\Domain\Contact\Contact;
+use Centreon\Domain\Contact\Interfaces\ContactInterface;
+use Centreon\Domain\Log\LoggerTrait;
+use Centreon\Domain\RequestParameters\Interfaces\RequestParametersInterface;
+use Centreon\Infrastructure\RequestParameters\RequestParametersTranslatorException;
+use Core\Application\Common\UseCase\ErrorResponse;
+use Core\Application\Common\UseCase\ForbiddenResponse;
+use Core\Command\Application\Repository\ReadCommandRepositoryInterface;
+use Core\Command\Domain\Model\Command;
+use Core\Command\Domain\Model\CommandType;
+use Core\Connector\Application\Exception\ConnectorException;
+use Core\Connector\Application\Repository\ReadConnectorRepositoryInterface;
+use Core\Connector\Domain\Model\Connector;
+
+final class FindConnectors
+{
+    use LoggerTrait;
+
+    public function __construct(
+        private readonly RequestParametersInterface $requestParameters,
+        private readonly ReadConnectorRepositoryInterface $readConnectorRepository,
+        private readonly ReadCommandRepositoryInterface $readCommandRepository,
+        private readonly ContactInterface $contact,
+    ) {
+    }
+
+    /**
+     * @param FindConnectorsPresenterInterface $presenter
+     */
+    public function __invoke(FindConnectorsPresenterInterface $presenter): void
+    {
+        try {
+            if (
+                ! $this->contact->hasTopologyRole(Contact::ROLE_CONFIGURATION_CONNECTORS_RW)
+                && ! $this->contact->hasTopologyRole(Contact::ROLE_CONFIGURATION_CONNECTORS_R)
+            ) {
+                $this->error(
+                    "User doesn't have sufficient rights to see connectors",
+                    ['user_id' => $this->contact->getId()]
+                );
+                $presenter->presentResponse(
+                    new ForbiddenResponse(ConnectorException::accessNotAllowed())
+                );
+
+                return;
+            }
+
+            $connectors = $this->readConnectorRepository->findByRequestParametersAndCommandTypes(
+                $this->requestParameters,
+                $this->retrieveConnectorTypesBasedOnContactRights(),
+            );
+
+            $commandIds = [];
+            foreach ($connectors as $connector)
+            {
+                array_push($commandIds, ...$connector->getcommandIds());
```

Comment:
Method case is not required in php but recommended.
```suggestion
                array_push($commandIds, ...$connector->getCommandIds());
```